logging/google/cloud/logging/_gax.py

```diff
@@ -603,8 +603,11 @@ def _log_entry_mapping_to_pb(mapping):
             entry_pb.labels[key] = value
 
     if 'jsonPayload' in mapping:
-        for key, value in mapping['jsonPayload'].items():
-            entry_pb.json_payload[key] = value
+        # NOTE: ``json.dumps`` is wasteful here because internally,
+        #       ``Parse`` will just call ``json.loads``. However,
+        #       there is no equivalent public function to parse on raw
+        #       dictionaries, so we waste cycles on parse/unparse.
```

Comment:
I'm assuming you already tried ParseDict? Looks like Parse just calls json.loads and then ParseDict so curious what's wrong with calling ParseDict directly.